### PR TITLE
use pureconfig to specify names of db designdocs

### PR DESCRIPTION
**Pull request by dgrove-oss**
Specify db design document names in application.conf
and support overrides via pureconfig conventions.

### CODE REVIEW

**Comment by dgrove-oss** [comment]
PG3 1593 🏃

**Comment by dgrove-oss** [comment]
PG2 2576 🏃

**Comment by dgrove-oss** [comment]
PG2 2576 🔵

**Comment by dgrove-oss** [comment]
Please kick travis build to try again -- PG clean, so very likely the 1 failure on travis is a flaky test case.

File: common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala [hunk 1]
```diff
@@ -147,13 +147,9 @@ object WhiskActivation
...
   override val collectionName = "activations"
 
-  // FIXME: reading the design doc from sys.env instead of a canonical property reader
-  // because WhiskConfig requires a logger, which requires an actor system, neither of
-  // which are readily available here; rather than introduce significant refactoring,
-  // defer this fix until WhiskConfig is refactored itself, which is planned to introduce
-  // type safe properties
-  private val mainDdoc = WhiskConfig.readFromEnv(dbActivationsDesignDoc).getOrElse("whisks.v2")
-  private val filtersDdoc = WhiskConfig.readFromEnv(dbActivationsFilterDesignDoc).getOrElse("whisks-filters.v2")
+  private val dbConfig = loadConfigOrThrow[DBConfig]("whisk.db")
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala]: should we have strings like `"whisk.db"` in a config singleton?

**Pull request merged**